modules/siddhi-core/src/main/java/org/wso2/siddhi/core/executor/function/IfThenElseFunctionExecutor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.siddhi.core.executor.function;
+
+import org.wso2.siddhi.core.config.ExecutionPlanContext;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+
+public class IfThenElseFunctionExecutor extends FunctionExecutor {
+
+    Attribute.Type returnType ;
+
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors,
+                        ExecutionPlanContext executionPlanContext) {
+        if (attributeExpressionExecutors.length != 3) {
+            // check whether all the arguments passed
+            throw new ExecutionPlanValidationException("Invalid no of arguments passed to ifThenElse() function, " +
+                    "required only 3, but found " + attributeExpressionExecutors.length);
+        }else if(!attributeExpressionExecutors[0].getReturnType().equals(Attribute.Type.BOOL)){
+            // check whether first argument Boolean or not
+            throw new ExecutionPlanValidationException("Input type of ifThenElse function executor " + attributeExpressionExecutors[0].toString() + " should be of type BOOL. " +
```

Comment:
Dont print as attributeExpressionExecutors[0].toString(), here use the attribute name  as write the exception as: 

"Input type of if in ifThenElse function should be of type BOOL, but found " + attributeExpressionExecutors[0].getReturnType()